lambda_function.py

```diff
@@ -1,218 +1,114 @@
+        self.run_command(cursor, '''SELECT /* Lambda CloudWatch Exporter */ 
+                                  \"schema\" || '.' || \"table\" AS table, 
+                                  encoded, max_varchar, unsorted, stats_off, tbl_rows, skew_sortkey1, skew_rows 
+                                  FROM svv_table_info''')
 
-if os.environ['db_name'] != None:
-    database = os.environ['db_name']
+        tables_not_compressed = 0
+        max_skew_ratio = 0
+        total_skew_ratio = 0
+        number_tables_skew = 0
+        number_tables = 0
+        max_skew_sort_ratio = 0
+        total_skew_sort_ratio = 0
+        number_tables_skew_sort = 0
+        number_tables_statsoff = 0
+        max_varchar_size = 0
+        max_unsorted_pct = 0
+        total_rows = 0
 
-if os.environ['cluster_name'] != None:
-    cluster = os.environ['cluster_name']
+        result = cursor.fetchall()
 
-try:
-    kms = boto3.client('kms')
-    password = kms.decrypt(CiphertextBlob=base64.b64decode(enc_password))['Plaintext']
-except:
-    print('KMS access failed: exception %s' % sys.exc_info()[1])
-    raise
+        for table in result:
+            table_name, encoded, max_varchar, unsorted, stats_off, tbl_rows, skew_sortkey1, skew_rows = table
+            number_tables += 1
+            if encoded == 'N':
+                tables_not_compressed += 1
+            if skew_rows is not None:
+                if skew_rows > max_skew_ratio:
+                    max_skew_ratio = skew_rows
+                total_skew_ratio += skew_rows
+                number_tables_skew += 1
+            if skew_sortkey1 is not None:
+                if skew_sortkey1 > max_skew_sort_ratio:
+                    max_skew_sort_ratio = skew_sortkey1
+                total_skew_sort_ratio += skew_sortkey1
+                number_tables_skew_sort += 1
+            if stats_off is not None and stats_off > 5:
+                number_tables_statsoff += 1
+            if max_varchar is not None and max_varchar > max_varchar_size:
+                max_varchar_size = max_varchar
+            if unsorted is not None and unsorted > max_unsorted_pct:
+                max_unsorted_pct = unsorted
+            if tbl_rows is not None:
+                total_rows += tbl_rows
 
-def run_external_commands(command_set_type, file_name, cursor, cluster):
-    if not os.path.exists(file_name):
-        return []
-    
-    external_commands = None
-    try:
-        external_commands = json.load(open(file_name, 'r'))
-    except ValueError as e:
-        # handle a malformed user query set gracefully
-        if e.message == "No JSON object could be decoded":
-            return []
-        else:
-            raise
-       
-    output_metrics = []
-    
-    for command in external_commands:
-        if command['type'] == 'value':
-            cmd_type = "Query"
+        if number_tables_skew > 0:
+            avg_skew_ratio = total_skew_ratio / number_tables_skew
```

Comment:
Do you want this to do Python's int division? Or should you convert one to a float?

Reply:
That is a good point, it is a ratio after all :)